### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by zawalich**
Improving model - longer training, and changing API host IP address to make it work seamlessly on minikube

### CODE REVIEW

**Comment by zawalich** [comment]
@faw01 one question for you - I am not Tensorflow expert:

Is there any way to create `class_names`/`label_to_class` variables without mapping classes from images directly by `data = tf.keras.utils.image_dataset_from_directory`? 

It would help trimming down Dockerfile significantly (now it needs adding sample images to work with API). 

I am basing my question on the content from here: [formula-one-image-classification.ipynb jupyter notebook](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb), lines:
- [127](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L127)
- [163](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L163)
- [91](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L91).

Let me know!

**Comment by faw01** [comment]
> @faw01 one question for you - I am not Tensorflow expert:
> 
> Is there any way to create `class_names`/`label_to_class` variables without mapping classes from images directly by `data = tf.keras.utils.image_dataset_from_directory`?
> 
> It would help trimming down Dockerfile significantly (now it needs adding sample images to work with API).
> 
> I am basing my question on the content from here: [formula-one-image-classification.ipynb jupyter notebook](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb), lines:
> 
> * [127](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L127)
> * [163](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L163)
> * [91](https://github.com/faw01/formula-one-image-classification-model/blob/main/formula-one-image-classification.ipynb?short_path=dfafb58#L91).
> 
> Let me know!

hi aleksander,

some context first: i referenced most of the original code from a binary image classification model, e.g. image is an apple or image is an orange (1 for apple, 0 for orange) - however the main issue with formula 1 having more than 2 teams would suggest that a binary model would not work in this case

hence, i applied a mapping to the classes

since we already know classes in out dataset (the teams), we can try manually defining them:
`class_names = ['alfa_romeo', 'bwt', 'ferrari', 'haas', 'mclaren', 'mercedes', 'redbull', 'renault', 'toro_rosso', 'williams']`

we would still need the mapping:
`label_to_class = {i: class_name for i, class_name in enumerate(class_names)}`

although, this allows us to define the class names and label mapping without needing to load images 

but i do recommend using:
`data = tf.keras.utils.image_dataset_from_directory('./input/formula-one-cars-images/train/')`

keras actually has a data pipeline direct function/helper built into it and we're accessing just that, and this just builds an image data set for us on the fly so we don't need to build labels, classes, etc - it will be automatic with a bunch of nice pre-processing out of the box such as image resizing, batches, etc

if it is an issue, alternatives are:
`tf.data.Dataset.list_files()` and `tf.io.read_file()`

**Comment by zawalich** [comment]
@faw01 thanks for clarification, it helped me severely! :)
